table/tables/index.go

```diff
@@ -198,8 +198,12 @@ func (c *index) GenIndexKey(sc *stmtctx.StatementContext, indexedValues []types.
 // Create will return the existing entry's handle as the first return value, ErrKeyExists as the second return value.
 func (c *index) Create(ctx sessionctx.Context, rm kv.RetrieverMutator, indexedValues []types.Datum, h int64,
 	opts ...*table.CreateIdxOpt) (int64, error) {
+	opt := &table.CreateIdxOpt{}
```

Comment:
How about

```
var skipUniqueIndexCheck bool
if len(opts) != 0 {
    skipUniqueIndexCheck = opts[0].SkipUniqueIndexCheck
}
...
skipCheck := ctx.GetSessionVars().LightningMode || ctx.GetSessionVars().StmtCtx.BatchCheck || skipUniqueIndexCheck
```

The allocation 
```
opt := &table.CreateIdxOpt{}
```
is unnecessary.

Reply:
I'm afraid there will be some other options in the future, then we will have to add more standalone variables.

Reply:
When more options is necessary in the future, we have to modify here again, right?
So, why not do that when it's really used, and take the better code style now?

Reply:
No need to modify here again, just add some flags at https://github.com/pingcap/tidb/blob/e06c87d3c1705dfd3557f77b502519c7445c5e6e/table/index.go#L31 , and use them at the right place.